contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -145,4 +156,212 @@ contract OETHVaultCore is VaultCore {
+
+        // store the next withdrawal request
+        withdrawalQueueMetadata.nextWithdrawalIndex = uint128(requestId + 1);
+        withdrawalQueueMetadata.queued = uint128(queued);
+
+        emit WithdrawalRequested(msg.sender, requestId, _amount, queued);
+
+        withdrawalRequests[requestId] = WithdrawalRequest({
+            withdrawer: msg.sender,
+            claimed: false,
+            amount: uint128(_amount),
+            queued: uint128(queued)
+        });
+    }
+
+    /**
+     * @notice Claim a previously requested withdrawal once it is claimable.
+     * @param requestId Unique ID for the withdrawal request
+     * @return amount Amount of WETH transferred to the withdrawer
+     */
+    function claimWithdrawal(uint256 requestId)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256 amount)
+    {
+        amount = _claimWithdrawal(requestId);
+
+        // transfer WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, amount);
+    }
+
+    /**
+     * @notice Claim a previously requested withdrawals once they are claimable.
+     * @param requestIds Unique ID of each withdrawal request
+     * @return amounts Amount of WETH received for each request
+     */
+    function claimWithdrawals(uint256[] memory requestIds)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256[] memory amounts, uint256 totalAmount)
+    {
+        amounts = new uint256[](requestIds.length);
+        for (uint256 i = 0; i < requestIds.length; ++i) {
+            amounts[i] = _claimWithdrawal(requestIds[i]);
+            totalAmount += amounts[i];
+        }
+
+        // transfer all the claimed WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, totalAmount);
+    }
+
+    // slither-disable-start reentrancy-no-eth
+
+    function _claimWithdrawal(uint256 requestId)
+        internal
+        returns (uint256 amount)
+    {
+        // Check if there's enough liquidity to cover the withdrawal request
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+        WithdrawalRequest memory request = withdrawalRequests[requestId];
+
+        require(request.claimed == false, "already claimed");
+        require(request.withdrawer == msg.sender, "not requester");
+
+        // Try and get more liquidity in the withdrawal queue if there is not enough
+        if (request.queued > queue.claimable) {
+            // Stream any harvested rewards (WETH) that are available to the Vault
+            IDripper(dripper).collect();
+
+            // Add any WETH from the Dripper to the withdrawal queue
+            uint256 addedClaimable = _addWithdrawalQueueLiquidity();
+
+            // If there still isn't enough liquidity in the queue to claim, revert
+            require(
+                request.queued <= queue.claimable + addedClaimable,
```

Comment:
Have you considered returning zero and letting the previous withdrawal execute in the case of a call to claimWithdrawals with one not being claimable in the array?

Reply:
I love reverts rather than silent ignores when something weird happens.